pages/recipe/[id].js

```diff
@@ -1,12 +1,14 @@
-import data from "@/db/data";
 import { useRouter } from "next/router";
 import RecipeDetails from "@/components/RecipeDetails.js";
 
-export default function RecipeDetailsPage() {
+export default function RecipeDetailsPage({ storedRecipes }) {
   const router = useRouter();
   const { id } = router.query;
+
   //Remember to remove the toString() because at the database there will be ids as strings
-  const recipeData = data.find((recipe) => recipe.id.toString() === id);
+  const recipeData = storedRecipes.find(
+    (recipe) => recipe.id.toString() === id
+  );
```

Comment:
Using string IDs in your data model could be a better practice than using numeric IDs, then you can save yourself the conversion to a string. 

Additionally, using string IDs in your data model would enhance its consistency. For creating new recipes, you are using UUIDs, which are inherently string-based.